esmvalcore/_data_finder.py

```diff
@@ -248,7 +292,7 @@ def _truncate_dates(date, file_date):
     elif len(date) > len(file_date):
         date = date[0:len(file_date)]
 
-    return date, file_date
+    return int(date), int(file_date)
```

Comment:
I am not sure about this because I had a bunch of issues with the parser and the input in `int`. So maybe this will break other parts of the run? Not sure though.

Reply:
Great then! Thanks for the explanation.